infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -1292,4 +1306,33 @@ public static int getChecksum(int num) {
             return hasher.putInt(num).hash().asInt();
         }
     }
+
+    /**
+     * Estimate the size (in bytes) of a directory.
+     * From https://stackoverflow.com/a/19869323
+     */
+    long estimateSize(Path directoryPath) {
+        final AtomicLong size = new AtomicLong(0);
```

Comment:
why AtomicLong?

Reply:
Accessing a long from a inner class would require it to be final, but then it must be immutable.